src/ViewRenderer/ViewRenderer.php

```diff
@@ -147,12 +144,43 @@ private function renderProxy(string $view, array $parameters = []): string
         );
     }
 
+    private function injectParameters(array $parameters, array $injections): array
+    {
+        foreach ($this->getInjections($injections) as $injection) {
+            $parameters = array_merge($parameters, $injection->getParams());
+        }
+        return $parameters;
+    }
+
+    /**
+     * @param array $injections
+     * @return InjectionInterface[]
+     */
+    private function getInjections(array $injections): array
+    {
+        $instances = [];
+        foreach ($injections as $key => $injection) {
+            if (is_string($injection)) {
+                $injection = $this->container->get($injection);
```

Comment:
This is a service locator approach. Maybe you should use `Reference::to()` in the web.php config instead.

Reply:
I see this here: https://github.com/yiisoft/yii-cycle/blob/861054745b61ddb9d3bbdea24fd07422ead56706/src/Schema/SchemaManager.php#L76

This allow comfortable set injections in params.

Reply:
Providers registration is preload stage